firebase-installations/src/main/java/com/google/firebase/installations/local/IidStore.java

```diff
@@ -86,14 +86,23 @@ private static String getDefaultSenderId(FirebaseApp app) {
     return projectNumber;
   }
 
-  private String createTokenKey(@NonNull String senderId) {
-    return STORE_KEY_TOKEN + senderId + DEFAULT_SCOPE;
+  private String createTokenKey(@NonNull String senderId, @NonNull String scope) {
+    return STORE_KEY_TOKEN + senderId + " | " + scope;
   }
 
   @Nullable
   public String readToken() {
     synchronized (iidPrefs) {
-      String token = iidPrefs.getString(createTokenKey(defaultSenderId), null);
+      Log.d("Rayo", "readToken: defaultSenderId: " + defaultSenderId);
+      String token = null;
+
+      for (String scope : ALLOWABLE_SCOPES) {
+        token = iidPrefs.getString(createTokenKey(defaultSenderId, scope), null);
+        if (!TextUtils.isEmpty(token)) {
+          break;
```

Comment:
How about we replace `break;` by this line:
```
return (token.startsWith(JSON_ENCODED_PREFIX)) ? parseIidTokenFromJson(token) : token;
```


And below instead of
```
if (TextUtils.isEmpty(token)) {
  return null;
}
```
We just do `return null;`

Reply:
done